src/hotspot/share/services/memJfrReporter.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_SERVICES_MEMJFRREPORTER_HPP
+#define SHARE_SERVICES_MEMJFRREPORTER_HPP
+
+#include "memory/allocation.hpp"
+#include "services/nmtUsage.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/ticks.hpp"
+
+// The MemJFRReporter and MemJFRCurrentUsage classes are only to be used from the
+// thread sending periodic JFR events. So no synchronization is needed.
+class MemJFRReporter : public AllStatic {
+private:
+  static void send_type_event(const Ticks& starttime, const char* tag, size_t reserved, size_t committed);
+ public:
+  static void send_total_event();
+  static void send_type_events();
+};
+
+// Helper class to avoid refreshing the NMTUsage to often and allow
+// the two JFR events to use the same data.
+class MemJFRCurrentUsage : public AllStatic {
```

Comment:
Move to .cpp file to because only used internally. Does not even have to be a class, can only be static fields and accessors.

Reply:
Moved it but kept the class structure to better show how things hang together :)